packages/block-library/src/image/edit.native.js

```diff
@@ -615,6 +606,10 @@ export class ImageEdit extends Component {
 		);
 	}
 
+	isGif( url ) {
+		return url.toLowerCase().includes( '.gif' );
```

Comment:
Not directly related to this PR, but is likely that this condition might overlook detecting some GIF images as not always the URL will contain the `.gif` value. 

I see the following potential alternatives for this:
1. Check if [`mime_type` attribute](https://developer.wordpress.org/rest-api/reference/media/#arguments) of the media item is `image/gif`. Note that this won't work if the URL provided is not a media item of the post, like the ones provided via the HTML of PR's testing instructions 1.
2. Prefetch the headers of the image and check the mime type. Note that this will require making an extra `HEAD` request for the image.
3. Once the image is cached into the disk, read the image attributes and check mime type.

In any case, as the current logic will probably cover most of the cases, I wouldn't consider it critical to provide a more accurate alternative.

Reply:
Fair point, and we actually went over [some ideas in the original PR too](https://github.com/WordPress/gutenberg/pull/38996#discussion_r812974538). We left it to the extra-simple-filename-check and to address this better in the future.

For the purposes of this new PR, I've only focused on expanding the flows the badge is shown, and haven't touched the actual detection logic at all. Hope we can do that soon.